app/backend/approaches/chatreadretrieveread.py

```diff
@@ -141,10 +141,10 @@ async def run(self, history: list[dict[str, str]], overrides: dict[str, Any]) ->
             system_message = prompt_override.format(follow_up_questions_prompt=follow_up_questions_prompt)
 
         messages = self.get_messages_from_history(
-            system_message + "\n\nSources:\n" + content,
+            system_message,
             self.chatgpt_model,
             history,
-            history[-1]["user"],
+            history[-1]["user"]+ "\n\nSources:\n" + content,
```

Comment:
It seems un-intuitive to me that sources would be part of a user message, so it'd help to add a comment here.

Reply:
done